src/Schema/Events/ImpressionTracking.ts

```diff
@@ -80,3 +80,28 @@ export interface TooltipViewed {
   context_owner_type: PageOwnerType
   type: string
 }
+
+/**
+ * A user sees a an error message
+ *
+ * This schema describes events sent to Segment from [[ErrorMessageViewed]].
+ *
+ *  @example
+ *  ```
+ *  {
+ *    action: "errorMessageViewed",
+ *    context_owner_type: "orders",
+ *    context_owner_id: "order-id",
+ *    label: "This is an error",
+ *    error_code: "charge_authorization_failed"
+ *  }
+ * ```
+ *
+ */
+export interface ErrorMessageViewed {
+  action: ActionType.errorMessageViewed
+  context_owner_id: string
+  context_owner_type: OwnerType
+  label: string
+  error_code: string
```

Comment:
We won't always have an error code when these errors occur. Could we make this optional and provide something like the name of the route where the error was raised?

Reply:
Is it better to use a 'generic_error_code' than a null if none is available?

Reply:
It would be a placeholder that we send instead of `null` when an error code is not available.

Reply:
Got it! We can leave null I think!